omegaconf/listconfig.py

```diff
@@ -86,6 +87,23 @@ def _validate_set(self, key: Any, value: Any) -> None:
                         "$FULL_KEY is not optional and cannot be assigned None"
                     )
 
+        from omegaconf import OmegaConf
+        from omegaconf._utils import is_attr_class, is_dataclass
+
+        target_type = self._metadata.element_type
```

Comment:
why not call it element_type?

Reply:
I called it target_type because in dictconfig's implementation of validate_set target_type. I'll change it to element_type if you prefer it.

Reply:
it's okay either way.